icechunk/src/change_set.rs

```diff
@@ -4,38 +4,39 @@ use std::{
     mem::take,
 };
 
+use bytes::Bytes;
 use itertools::{Either, Itertools as _};
 use serde::{Deserialize, Serialize};
 
 use crate::{
     format::{
         manifest::{ChunkInfo, ChunkPayload},
-        snapshot::{NodeData, NodeSnapshot, UserAttributesSnapshot, ZarrArrayMetadata},
+        snapshot::{ArrayShape, DimensionName, NodeData, NodeSnapshot},
         ChunkIndices, NodeId, Path,
     },
-    metadata::UserAttributes,
     session::SessionResult,
 };
 
-#[derive(Clone, Debug, PartialEq, Default, Serialize, Deserialize)]
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct ArrayData {
```

Comment:
Wondering why not share this with `NodeData::Array` like before. But I guess this struct is so small, that flatter is nicer.

Reply:
I thought it wasn't 100% the same? I think user_data is in different places?